cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -264,9 +263,26 @@ public void recordProgramStop(Id.Program program, String pid, long stopTs, Progr
   // Any changes made here will have to be made over there too.
   // JIRA https://issues.cask.co/browse/CDAP-2172
   public RunRecord getRun(Id.Program program, final String runid) {
-    // For querying running records
+    // Query active run record first
+    RunRecord running = getNonCompleteRun(program, TYPE_RUN_RECORD_STARTED, runid);
+    // If program is running, this will be non-null
+    if (running != null) {
+      return running;
+    }
+
+    // If program is not running, query completed run records
+    RunRecord complete = getCompletedRun(program, runid);
+    if (complete != null) {
+      return complete;
+    }
+
+    // Else query suspended run records
+    return getNonCompleteRun(program, TYPE_RUN_RECORD_SUSPENDED, runid);
+  }
+
+  private RunRecord getNonCompleteRun(Id.Program program, String recordType, String runid) {
```

Comment:
Should we add one line comment saying what is included in the `NonComplete` run?